react-frontend/src/profile/ProfileDetails.js

```diff
@@ -1,20 +1,56 @@
-import React from "react";
+import React, {useState} from "react";
+import axios from 'axios';
+import PropTypes from "prop-types";
+import Notify from '../components/ErrorNotification';
 
-function ProfileDetails() {
+function ProfileDetails(props) {
+    const [bio, setBio] = useState("");
+    const [showNotification, setShowNotification] = useState(false);
+    const [notificationMsg, setNotificationMsg] = useState("");
+    const [isSuccessful, setIsSuccessful] = useState(false);
 
+    function handleChange(event) {
+        setBio(event.target.value);
+    }
+
+    const saveBio = async() => {
+        try {
+            await axios.post(`${process.env.REACT_APP_BACKEND_LINK}/users/set-bio`, {
+                userID: props.userID,
+                bio: bio
+            }, {
+                headers: {
+                    'Authorization': `Bearer ${localStorage.getItem('token')}`
+                  }
+            });
+            setNotificationMsg("Bio Saved Successfully");
+            setIsSuccessful(true);
+            setShowNotification(true);
+        } catch (error) {
+            console.error("Error:", error);
+            setIsSuccessful(false);
+            setNotificationMsg("Save Unsuccessful");
+            setShowNotification(true);
+        }
+    }
 
     return (
         <div className = "padding-left">
         <div className = "small-container drop-shadow profile-height">
             <h5>Profile Details</h5>
-            <textarea className="vertical-form" placeholder = "Add your bio here.."></textarea>
-            <button className = "small-button">Save Bio</button>
+            <textarea className="vertical-form" placeholder = "Add your bio here.." value ={bio} onChange={handleChange}></textarea>
```

Comment:
> Might be better to just use a < form > and set height using style={} instead of vertical-form with a textarea. Not that different but would remove the need for an extra css style in index.css

Gotcha on the styling. As for the input type, <form> is for single-line user inputs, if I set it to a form all text will be centered vertically in the input box.